platform/source/mbed_retarget.cpp

```diff
@@ -768,13 +866,31 @@ extern "C" int PREFIX(_read)(FILEHANDLE fh, unsigned char *buffer, unsigned int
 
 extern "C" ssize_t read(int fildes, void *buf, size_t length)
 {
+    if (length == 0) {
+        return 0;
+    }
+
+#if MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY && DEVICE_SERIAL
```

Comment:
`DEVICE_SERIAL` is not relevant here. Can have a minimal non-serial console on a chip with no serial.

Check same for `write`.